base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1966,22 +2019,23 @@ fn swap_to_highest_pow_chain<T: BlockchainBackend>(
     // rewind to height will first delete the headers, then try delete from blocks, if we call this to the current
     // height it will only trim the extra headers with no blocks
     rewind_to_height(db, metadata.height_of_longest_chain())?;
-    let all_orphan_tips = db.fetch_all_orphan_chain_tips()?;
-    if all_orphan_tips.is_empty() {
+    let strongest_orphan_tips = db.fetch_strongest_orphan_chain_tips()?;
+    if strongest_orphan_tips.is_empty() {
         // we have no orphan chain tips, we have trimmed remaining headers, we are on the best tip we have, so lets
         // return ok
         return Ok(BlockAddResult::OrphanBlock);
```

Comment:
See previous comment, but this is a strange thing to return IMO

Reply:
I think its the correct thing to return. 
If all the orphans tips are empty. It means the block you added is an orphan, hence the result here. 
New blocks are first added as an orphan, and then from there, an orphan chain tip is created if you can link it to the main chain. Then do swap to the highest chain. So any new block will always be an orphan chain tip first.

Reply:
I agree with @SWvheerden on this

Reply:
I don't think you understand my point. The method name "swap_to_highest_chain" suggests it can be called at any time, but the value returned is tightly coupled to the caller context.... i.e. it is returning a critical result that the caller "handle_possible_reorg" requires. this is a private method, so it's probably fine, but it's a code smell that could potentially cause a bug in future.